src/batch/src/executor/log_row_seq_scan.rs

```diff
@@ -0,0 +1,277 @@
+                Arc::new(schema.clone()),
+            );
+            #[for_await]
+            for chunk in stream {
+                let chunk = chunk?;
+                yield chunk;
+            }
+        }
+    }
+
+    #[try_stream(ok = DataChunk, error = BatchError)]
+    async fn execute_range(
+        table: Arc<StorageTable<S>>,
+        scan_range: ScanRange,
+        old_epoch: BatchQueryEpoch,
+        new_epoch: BatchQueryEpoch,
+        chunk_size: usize,
+        histogram: Option<impl Deref<Target = Histogram>>,
+        schema: Arc<Schema>,
+    ) {
+        let ScanRange {
+            pk_prefix,
+            next_col_bounds,
+        } = scan_range;
+
+        let order_type = table.pk_serializer().get_order_types()[pk_prefix.len()];
+        let (start_bound, end_bound) = (next_col_bounds.0, next_col_bounds.1);
+        assert!(pk_prefix.len() < table.pk_indices().len());
+        // Range Scan.
+        let iter = table
+            .batch_iter_log_with_pk_bounds(
+                old_epoch.into(),
+                new_epoch.into(),
+                &pk_prefix,
+                (
+                    match start_bound {
+                        Bound::Unbounded => {
+                            if order_type.nulls_are_first() {
+                                // `NULL`s are at the start bound side, we should exclude them to meet SQL semantics.
+                                Bound::Excluded(OwnedRow::new(vec![None]))
+                            } else {
+                                // Both start and end are unbounded, so we need to select all rows.
+                                Bound::Unbounded
+                            }
+                        }
+                        _ => unimplemented!("Log iter range need full"),
+                    },
+                    match end_bound {
+                        Bound::Unbounded => {
+                            if order_type.nulls_are_last() {
+                                // `NULL`s are at the end bound side, we should exclude them to meet SQL semantics.
+                                Bound::Excluded(OwnedRow::new(vec![None]))
+                            } else {
+                                // Both start and end are unbounded, so we need to select all rows.
+                                Bound::Unbounded
+                            }
+                        }
+                        _ => unimplemented!("Log iter range need full"),
+                    },
+                ),
+            )
+            .await?;
+
+        pin_mut!(iter);
+        loop {
+            let timer = histogram.as_ref().map(|histogram| histogram.start_timer());
+
+            let mut iter = iter.as_mut().map(|r| match r {
+                Ok((op, value)) => {
+                    let (k, row) = value.into_owned_row_key();
+                    let full_row = row
+                        .into_iter()
+                        .chain(vec![Some(ScalarImpl::Int16(op))])
```

Comment:
May leave a TODO here to avoid create a full row (which is vec) again to avoid memory allocation again. Currently we need to create a new `OwnedRow` because `collect_data_chunk` only accept streams with `KeyedRow`, but `KeyedRow` can only accept `OwnedRow` as row type.